batch/batch/cloud/gcp/instance_config.py

```diff
@@ -88,6 +88,8 @@ def worker_type(self) -> str:
 
     def region_for(self, location: str) -> str:
         # location = zone
+        if location.startswith('projects'):
```

Comment:
What's this "projects" about?

Reply:
The issue is the return value from using Google's metadata server is a path and not just the zone name. It's like `/projects/hail-vdc/random-hash/us-central1-a`. I can either pass the zone directly that we know when creating the instance into the startup script or I can try and do a better job integrating the parsing with the other place in the code base where we have to parse the ZONE that starts with `projects`. Preferences?

Reply:
ah ya let's just use the custom userdata for this like in your first commit and not use the metadata server's endpoint.